examples/06-threading.rs

```diff
@@ -0,0 +1,28 @@
+use cudarc::driver::{CudaDevice, DeviceSlice, DriverError};
+
+use std::thread;
+
+fn main() -> Result<(), DriverError> {
+    let dev = CudaDevice::new(0)?;
+
+    let dev1 = dev.clone();
+    let thread1 = thread::spawn(move || {
+        // You'll need to call bind_to_thread on each thread that the
+        // device is used on.
+        dev1.bind_to_thread()?;
+        dev1.alloc_zeros::<f32>(100)
```

Comment:
Should launch a kernel here that prints out a message to show that things are sequentially executed.